Cql/Cql.Compiler/ExpressionBuilder.cs

```diff
@@ -145,63 +145,110 @@ public DefinitionDictionary<LambdaExpression> Build()
-                        .ToDictionary(cs => cs.name, cs => cs.id)
-                        ?? new Dictionary<string, string>();
-
-                    var codesByCodeSystemName = new Dictionary<string, List<CqlCode>>();
-                    var codeCtor = typeof(CqlCode).GetConstructor(new Type[]
-                    {
-                        typeof(string),
-                        typeof(string),
-                        typeof(string),
-                        typeof(string)
-                    })!;
-
                     foreach (var code in Library.codes)
                     {
                         if (code.codeSystem == null)
                             throw new InvalidOperationException("Code definition has a null codeSystem node.");
+                        if (!codeSystemUrls.TryGetValue(code.codeSystem.name, out var csUrl))
+                            throw new InvalidOperationException($"Undefined code system {code.codeSystem.name!}");
+                        var existingCode = codesByName.Values.SingleOrDefault(c => c.code == code.id && c.system == csUrl);
+                        if (existingCode != null)
+                            throw new InvalidOperationException($"Duplicate code detected: {code.id} from {code.codeSystem.name} ({csUrl})");
+                        var systemCode = new CqlCode(code.id, csUrl, null, null);
+                        codesByName.Add(code.name, systemCode);
                         if (!codesByCodeSystemName.TryGetValue(code.codeSystem!.name!, out var codings))
                         {
                             codings = new List<CqlCode>();
                             codesByCodeSystemName.Add(code.codeSystem!.name!, codings);
                         }
-                        if (!codeSystemUrls.TryGetValue(code.codeSystem.name, out var csUrl))
-                            throw new InvalidOperationException($"Undefined code system {code.codeSystem.name!}");
-                        var existingCode = codings.SingleOrDefault(c => c.code == code.id && c.system == csUrl);
-                        if (existingCode != null)
-                            throw new InvalidOperationException($"Duplicate code detected: {code.id} from {code.codeSystem.name} ({csUrl})");
-                        codings.Add(new CqlCode(code.id, csUrl, null, null));
-
+                        codings.Add(systemCode);
 
                         var newCodingExpression = Expression.New(codeCtor,
                             Expression.Constant(code.id),
                             Expression.Constant(csUrl),
                             Expression.Constant(null, typeof(string)),
                             Expression.Constant(null, typeof(string))!
                         );
-
                         var contextParameter = Expression.Parameter(typeof(CqlContext), "context");
                         var lambda = Expression.Lambda(newCodingExpression, contextParameter);

```

Comment:
This lambda should create a `CqlConcept` (and include the description!), because that's what this construct is creating. If we are allowed to use that in the RHS of the retrieve, then we should let CqlConcept implement "list of codes".

Reply:
Implemented & tested